docs/source/extensions_tutorial/extension_examples/labmetadata_extension.rst

```diff
@@ -0,0 +1,347 @@
+     or :py:class:`~pynwb.file.ElectrodeTable` to describe extracellular electrodes. Depending on the
+     type of metadata, use of these existing dynamic table structures can help avoid the need for
+     custom extensions by including the data as additional, custom columns in the appropriate existing tables.
+
+Creating the extension
+""""""""""""""""""""""
+
+**1.** Create a new repository for the extension using the :nwb_extension_git:`ndx-template`:
+
+.. code-block:: bash
+
+   cookiecutter gh:nwb-extensions/ndx-template
+
+**2.** Answer a few simple questions of the cookiecutter template. We can respond to many questions
+with ``Enter`` to accept the default response (e.g., to start with ``version=0.1.0``):
+
+.. code-block::
+
+    namespace [ndx-my-namespace]: ndx-labmetadata-example
+    description [My NWB extension]: Example extension to illustrate how to extend LabMetaData for adding lab-specific metadata
+    author [My Name]: Oliver Ruebel
+    email [my_email@example.com]: oruebel@lbl.gov
+    github_username [myname]: oruebel
+    copyright [2021, Oliver Ruebel]:
+    version [0.1.0]:
+    release [alpha]:
+    Select license:
+    1 - BSD-3
+    2 - MIT
+    3 - Apache Software License 2.0
+    4 - Other
+    Choose from 1, 2, 3, 4 [1]: 1
+    py_pkg_name [ndx_labmetadata_example]:
+
+**3.** Edit ``ndx-my-brainlabsrc/spec/create_extension_spec.py`` that was generated for you to define the
+schema of your extension. See the :ref:`extension-spec-api` section for details on how to use the specification API.
+
+* Add ``LabMetaData`` as an include type:
+
+.. code-block:: python
+
+    ns_builder.include_type('LabMetaData', namespace='core')
+
+* Define your new ``LabMetaData`` type for your lab
+
+.. code-block:: python
+
+     labmetadata_ext = NWBGroupSpec(
+        name='custom_lab_metadata',
+        doc='Example extension type for storing lab metadata',
+        neurodata_type_def='LabMetaDataExtensionExample',
+        neurodata_type_inc='LabMetaData',
+    )
+
+* Add the ``Groups``, ``Datasets``, and ``Attributes`` with the metadata specific to our lab to
+  our ``LabMetaData`` schema
+
+.. code-block:: python
+
+    labmetadata_ext.add_dataset(
+        name="tissue_preparation",
+        doc="Lab-specific description of the preparation of the tissue",
+        dtype='text',
+        quantity='?'
+    )
+
+* Add our new type definitions to the extension
+
+.. code-block:: python
+
+    new_data_types = [labmetadata_ext]
+
+**4.** Generate the schema for the extension by running the ``create_extension_spec.py`` script
+
+.. code-block:: bash
+
+   cd ndx-labmetadata-example
+   python src/spec/create_extension_spec.py
+
+**5.** To make sure our schema files are tracked by Git, we now add the YAML schema files we just created to the repo:
+
+.. code-block:: bash
+
+    git add spec/*.yaml
+    git commit -m "Added extension schema files"
+
```

Comment:
Why not add all the files?

Reply:
Do you mean do ``git add spec/`` instead? Since the spec folder should only contain YAML files, both approaches should typically be identical. Either way is fine.

Reply:
No, I mean add the Python files, i.e. `git add .` instead. This could be a separate step.